public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.ts

```diff
@@ -351,43 +351,63 @@ export async function getAfterSelectorCompletions(
 export async function getLogfmtCompletions(
   logQuery: string,
```

Comment:
if all of these args are properties on the situation object (besides dataProvider), do we want to keep arg-stuffing this function or should we just pass in the situation to this function?

Reply:
At least in this PR, yes. I would love to receive situations at some point, but because of how situations have been [defined](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/situation.ts#L100-L146), the compiler has a hard time deciding which situation it is.

![imagen](https://github.com/grafana/grafana/assets/1069378/b100d130-1ea0-4cca-b4ba-71985c4710bb)

I've recently discussed with the squad that, whenever we have time, I would really like to go back to this implementation and do a round of clean up of the original implementation.